src/dex/smardex/smardex.ts

```diff
@@ -0,0 +1,842 @@
+    blockNumber: number,
+  ): Promise<SmardexPoolState[]> {
+    try {
+      const multiCallFeeData = pairs.map(pair =>
+        this.getFeesMultiCallData(pair.exchange!),
+      );
+      const calldata = pairs
+        .map((pair, i) => {
+          let calldata = [
+            {
+              target: pair.exchange!,
+              callData: smardexPoolL1.encodeFunctionData('getReserves'),
+            },
+            {
+              target: pair.exchange!,
+              callData: smardexPoolL1.encodeFunctionData('getFictiveReserves'),
+            },
+            {
+              target: pair.exchange!,
+              callData: smardexPoolL1.encodeFunctionData('getPriceAverage'),
+            },
+          ];
+          // Fetch fees only on layer 2
+          !this.isLayer1() && calldata.push(multiCallFeeData[i]!.callEntry);
+          return calldata;
+        })
+        .flat();
+
+      const data: { returnData: any[] } =
+        await this.dexHelper.multiContract.methods
+          .aggregate(calldata)
+          .call({}, blockNumber);
+
+      const returnData = _.chunk(data.returnData, 4);
+      return pairs.map((_pair: SmardexPair, i) => ({
+        reserves0: coder
+          .decode(['uint256', 'uint256'], returnData[i][0])[0]
+          .toString(),
+        reserves1: coder
+          .decode(['uint256', 'uint256'], returnData[i][0])[1]
+          .toString(),
+        fictiveReserves0: coder
+          .decode(['uint256', 'uint256'], returnData[i][1])[0]
+          .toString(),
+        fictiveReserves1: coder
+          .decode(['uint256', 'uint256'], returnData[i][1])[1]
+          .toString(),
+        priceAverage0: coder
+          .decode(['uint256', 'uint256', 'uint256'], returnData[i][2])[0]
+          .toString(),
+        priceAverage1: coder
+          .decode(['uint256', 'uint256', 'uint256'], returnData[i][2])[1]
+          .toString(),
+        priceAverageLastTimestamp: coder
+          .decode(['uint256', 'uint256', 'uint256'], returnData[i][2])[2]
+          .toString(),
+        feesLP: this.isLayer1()
+          ? FEES_LAYER_ONE.feesLP
+          : multiCallFeeData[i]!.callDecoder(returnData[i][3]).feesLP,
+        feesPool: this.isLayer1()
+          ? FEES_LAYER_ONE.feesPool
+          : multiCallFeeData[i]!.callDecoder(returnData[i][3]).feesPool,
+      }));
+    } catch (e) {
+      this.logger.error(
+        `Error_getManyPoolReserves could not get reserves with error:`,
+        e,
+      );
+      return [];
+    }
+  }
+
+  protected fixPath(path: Address[], srcToken: Address, destToken: Address) {
+    return path.map((token: string, i: number) => {
+      if (
+        (i === 0 && srcToken.toLowerCase() === ETHER_ADDRESS.toLowerCase()) ||
+        (i === path.length - 1 &&
+          destToken.toLowerCase() === ETHER_ADDRESS.toLowerCase())
+      )
+        return ETHER_ADDRESS;
+      return token;
+    });
+  }
+
+  // Necessary to get the correct path for the router
+  getDirectParam(
```

Comment:
@JackieJoo Not sure we need this method since I'm not aware about direct method support for Smardex on our contract side

Reply:
fixed